lib/spack/llnl/util/lock/dummy.py

```diff
@@ -0,0 +1,48 @@
+##############################################################################
+# Copyright (c) 2013-2018, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+"""Implementation of ``llnl.util.lock`` that does no locking.
+
+You can import this package instead of llnl.util.lock to disable locking
+in your code.
+"""
+import llnl.util.lock
+from llnl.util.lock import *  # noqa
```

Comment:
An alternative way of structuring the code could be:
```
lock/
    __init__.py # almost empty, selects the implementation based on user's request and exports classes
    _flock.py # current implementation
    _dummy.py # dummy implementation
```
Both implementations are fine with me, just want to point out this different layout if you didn't already consider it.

Reply:
I think I like the current one b/c you explicitly import dummy, and the details of selecting an implementation are left up to `spack.util.lock`.  How would the importer control which version is imported in the layout above?